### ISSUE TITLE
Full width layout support

### ISSUE THREAD

**Issue by davidsherlock** [easydigitaldownloads/vendd#203]
I am currently in the process of developing a child theme for Vendd and want to thank you @SDavisMedia for providing such a useful base theme/framework. Really a great starting point to building a more comprehensive marketplace. Additionally, I plan on selling this child theme once complete.

I do however have a few suggestions (which I will post as separate issues here - apologies for the volume of them). The first one is the ability to have full width theme rather than the boxed layout, where at least the header takes up the full browser width. Ideally this setting would be within the customizer.

**Comment by SDavisMedia** [comment]
@davidsherlock thanks for the kind words! I've actually considered that. I built that exact option in other themes of mine. I even had plans to do it in Vendd at one point: https://github.com/easydigitaldownloads/vendd/issues/113

The reason I decided not to do honestly comes down to return on investment. As you know, our themes are free. We still have to support them, though. We don't get overloaded with support at for them at the moment but it's enough to make me think about ramifications of more features and more development time.

That said, I'm definitely willing to revisit this. From a visual standpoint, I assume you're imagining the Info Bar, Header (and its menu), and the Footer to span full-width with its content still within the same page dimensions in place now.

**Comment by davidsherlock** [comment]
I'm actually developing two commercial child themes for Vendd at the minute, one based on Google material design standards and a sleek dark theme for modern digital stores (photography, music, etc.). I'm adding a lot of options and features myself but wanted to suggest a few for the parent theme to avoid bloat and duplication.

I should note that I agree with 99% of the design and code decisions you've made and the reason I like the theme is due to the lightweight, non-bloated nature and ease of development. I plan to keep it that way with the child themes as well, which hopefully are well received by the EDD team.

Personally speaking I feel the "boxed" layout looks and feels a bit "affiliate marketing blog" in nature and having the ability to switch layouts globally within the customiser would be excellent. Ultimately the main content body can be extended easily and background colours adjusted to make it appear seamless, but having a full width header section would make it appear more modern and I'll handle the styling stuff.

Hopefully you can squeeze this into a future release. Are you happy for me to suggest further minor tweaks here on GitHub as I develop at my side?

**Comment by SDavisMedia** [comment]
@davidsherlock I'm always open to suggestions, no doubt about that. 

I'll see about getting a full-width structure added. Should be simple enough to implement.

**Comment by davidsherlock** [comment]
Thanks @SDavisMedia, I'll send you the child themes once they are complete for review if you like?

**Comment by SDavisMedia** [comment]
@davidsherlock absolutely! Wouldn't mind seeing what you come up with.

**Comment by SDavisMedia** [comment]
@davidsherlock I'm gonna try to get it in the next release. If it's feasible, the other issues will be bumped up to 1.2 and they'll all be in the next update.

**Comment by SDavisMedia** [comment]
I've almost got this done. It'll be in the next release for sure.

**Comment by davidsherlock** [comment]
Legend! Thanks Sean. I've actually started developing the new (forked) theme based on Vendd and it's shaping up nicely. 

Do you have a rough idea when next version will be out? No rush, just curious as I'm building this new theme.

**Comment by SDavisMedia** [comment]
@davidsherlock probably this week. Everything I wanted in 1.1.11 is done. Trying to decide if I wanna drop this issue down to 1.1.11 and release, or bump the others up to 1.2. If I bump the others up, I'll end up combing through all the other issues to see what else I can squeeze in just to make 1.2 a little more significant.

**Comment by SDavisMedia** [comment]
@davidsherlock check that out. Let me know what you think. It's done mainly with CSS. 

If you're familiar with full-width layouts already, I'm sure you're used to seeing multiple stacked HTML elements and you basically just put them where you want them in the HTML flow between your header and footer. Because Vendd was built to have a page-width structure, the opening tag for the content area is in the header.php file and its closing tag is in the footer.php. That means that even though it's now a full-width, anything anyone places in the content area, custom or otherwise, is forced into that content element of the "stack."

For 99.99% of users, they don't care. For anyone doing really heavy customizations, they could eventually hit that roadblock. I had to decide between 1) adding more PHP (would need conditionals) to my user-facing template files to make life easier on the .01% or 2) keep the current setup since it's an easy enough job for the .01% to override templates from a child theme if they really need to get creative. I chose 2. I think that's the right choice for Vendd.
